ast2vec/model2/proxbase.py

```diff
@@ -135,17 +111,37 @@ def fill_mat(item_to_item, offset):
         fill_mat(roles_to_roles, (0, 0))
         fill_mat(roles_to_tokens, (0, n_roles))
         fill_mat(tokens_to_tokens, (n_roles, n_roles))
+
         mat = coo_matrix(mat + mat.T - diags(mat.diagonal()))
         tokens, mat = self._adj_to_feat(self.role2ind, self.token2ind, mat)
+        self._clear()
 
         prox = Cooccurrences()
         prox.construct(tokens=tokens, matrix=mat)
-        if destdir.endswith(".asdf"):
-            name = destdir
-        else:
-            name = os.path.join(
-                destdir, "%s2%s_%d.asdf" % (UASTModel.NAME, Cooccurrences.NAME, index))
-        prox.save(name)
+        return prox
+
+    def _adj_to_feat(self, role2ind: Dict[int, int], token2ind: Dict[int, int], mat) -> Tuple:
+        """
+        This must be implemented in the child classes.
+
+        :param role2ind: Mapping from roles to indices, starting with 0.
+        :param token2ind: Mapping from tokens to indices, starting with 0.
+        :param mat: Adjacency matrix ('scipy.sparse.coo_matrix') with rows corresponding to
+                    node roles followed by node tokens.
+        :return: tuple('tokens', 'matrix'). 'tokens' are generalized tokens (usually roles+tokens).
+                 'matrix' rows correspond to 'tokens'.
+        """
+        raise NotImplementedError
+
+    def _clear(self):
+        """
+        Release memory.
```

Comment:
You call this from __init__ and apparently do not release any memory in that case. Rephrase